### PR TITLE
Filtre data tree

### PR DESCRIPTION
**Pull request by stafyniaksacha**
Add search into data tree

todo: 
- [x] tests

### CODE REVIEW

**Comment by xbill82** [comment]
Amazing!
But there's a strange behavior, here's to reproduce:
* type something in the search bar
* click on one of the resulting collections
* the magnifier icon in the search bar disappears.

**Comment by xbill82** [comment]
I moved the `highlight` filter to the `IndexBranch.vue` component scope, which I find better than globally registering the filter in `main.js`. Is this OK for you?

**Comment by xbill82** [comment]
Once we agree on the filter and the icon-bug is solved, I approve this PR. :+1:

**Comment by stafyniaksacha** [comment]
filter and the icon-bug is solved ;)

File: src/components/Data/Browse/IndexBranch.vue [hunk 1]
```diff
@@ -89,6 +95,9 @@ export default {
   watch: {
     index: function (index) {
       this.isTreeOpen(index, this.indexTree.name)
+    },
+    collection: function () {
```
Comment (AnthonySendra) [inline src/components/Data/Browse/IndexBranch.vue]: Back to 90"
```
collection () {
}
```

File: src/filters/highlight.filter.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export function highlight (value, filter) {
+  if (value && value !== '') {
+    let index = value.indexOf(filter)
+
+    if (index >= 0) {
+      value = value.substring(0, index) +
+        '<strong class="highlight">' +
```
Comment (AnthonySendra) [inline src/filters/highlight.filter.js]: No realy fan about adding HTML in a filter with a specific application CSS class `highlight`. But if we don't have the choice, ok.

File: test/unit/specs/data/browse/treeview.spec.js [hunk 3]
```diff
@@ -1,32 +1,133 @@
...
+      }
+    }
+
+    const App = Vue.extend({
+      template: '<div><router-view v-ref:routerview></router-view></div>',
+      replace: false
+    })
+
+    const TestComponent = Vue.extend({
+      template: '<treeview ' +
```
Comment (AnthonySendra) [inline test/unit/specs/data/browse/treeview.spec.js]: What about

```
`<treeview
  v-ref:treeview 
  :index="index" 
  :collection="collection" 
  :index-tree="tree">
</treeview>`
```

File: src/filters/highlight.filter.js (hunk 2 above)
Comment (stafyniaksacha) [inline src/filters/highlight.filter.js]: If you have a better way, I'll take it :)

**Comment by AnthonySendra** [comment]
:+1:

**Pull request merged**